salt/state.py

```diff
@@ -757,7 +758,12 @@ def __init__(
         loader="states",
         initial_pillar=None,
         file_client=None,
+        invocation_id=None,
```

Comment:
I recommend adding a docstring that describes the purpose of this `invocation_id` and what value should be passed to it. Specifically describe the difference between `invocation_id` and `jid` and the intended differences in use cases. Also should this even be an argument or only set internally in the `__init__`--it looks like nothing is currently passing in the argument?

Reply:
It should never be passed when instantiated externally and in most cases must be equal to `jid`. It's an init arg because on spawning platforms the object is reconstructed here:

https://github.com/saltstack/salt/blob/551e7c4b0456ca520c04cb2b8c7d3030b34afdce/salt/state.py#L2151-L2152

Agreed about the docstring. Also, if `jid` is not None, it should always be forced to be equal to it, need to change that logic.